lib/tesla_curl.ex

```diff
@@ -201,42 +144,84 @@ defmodule Tesla.Middleware.Curl do
 
   defp parse_body(body, flag_type, opts) do
     Enum.flat_map(body, fn {k, v} ->
-      results = translate_value(k, v)
-      Enum.map(results, fn {key, value} -> filter_body(flag_type, key, value, opts) end)
+      translate_parameters(flag_type, k, v, opts)
     end)
     |> Enum.join(" ")
     |> Kernel.<>(" ")
   end
 
-  # Recursively handles any nested maps or lists, returns a tuple containing the translated keys and values in string form.
-  @spec translate_value(String.t(), map() | list() | String.t()) :: [{String.t(), String.t()}]
-  defp translate_value(key, value) when is_map(value) do
-    value
+  # Recursively handles any nested maps or lists, returns a list of the translated parameters
+  @spec translate_parameters(String.t(), String.t(), any(), keyword() | nil) :: [String.t()]
+  defp translate_parameters(flag_type, key, value, opts) when is_map(value) do
+    maybe_redact_field(key, value, opts)
     |> Map.to_list()
     |> Enum.flat_map(fn {k, v} ->
-      translate_value("#{key}[#{k}]", v)
+      translate_parameters(flag_type, "#{key}[#{k}]", v, opts)
     end)
   end
 
-  defp translate_value(key, value) when is_tuple(value) do
-    value
+  defp translate_parameters(flag_type, key, value, opts) when is_tuple(value) do
+    maybe_redact_field(key, value, opts)
     |> Enum.flat_map(fn {k, v} ->
-      translate_value("#{key}[#{k}]", v)
+      translate_parameters(flag_type, "#{key}[#{k}]", v, opts)
     end)
   end
 
-  defp translate_value(key, value) when is_list(value) do
-    value
+  defp translate_parameters(flag_type, key, value, opts) when is_list(value) do
+    maybe_redact_field(key, value, opts)
     |> Enum.with_index()
     |> Enum.flat_map(fn {v, i} ->
-      translate_value("#{key}[#{i}]", v)
+      translate_parameters(flag_type, "#{key}[#{i}]", v, opts)
+    end)
+  end
+
+  defp translate_parameters(flag_type, key, value, opts) do
+    safe_value = maybe_redact_field(key, value, opts)
+    [construct_parameter(flag_type, key, safe_value)]
+  end
+
+  # Checks if the key matches any of the redact_fields, including ones found in nested maps or lists
+  @spec field_needs_redaction(String.t(), list()) :: list()
+  defp field_needs_redaction(key, redact_fields) do
+    downcased_key = String.downcase(key)
+
+    Enum.map(redact_fields, fn field ->
+      if !Regex.regex?(field) do
+        downcased_field = String.downcase(field)
+
+        String.contains?(downcased_key, downcased_field) ||
+          String.contains?(downcased_key, "[#{downcased_field}]")
+      end
     end)
   end
 
-  defp translate_value(key, value) do
-    [{key, value}]
+  # Redacts the value if the key matches any of the redact_fields, if supplied
+  @spec maybe_redact_field(String.t(), any(), keyword() | nil) :: any()
+  defp maybe_redact_field(_key, value, nil), do: value
+  defp maybe_redact_field(_key, value, _opts) when is_map(value), do: value
```

Comment:
Maybe we want to redact all children values? E.g. if `wiki_page` is supplied and the params are , then it would `%{wiki_page: %{name: "Some Name", is_published: true}}`be`%{wiki_page: %{name: "[REDACTED]", is_published: "[REDACTED]"}}`